### PR TITLE
Enhancement: Enable header_comment fixer

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] enables the `header_comment` fixer
* [x] runs `composer assert:cs-fix`

### CODE REVIEW

File: .php_cs [hunk 1]
```diff
@@ -1,16 +1,78 @@
...
+If you did not receive a copy of the license and are unable to
+obtain it through the world-wide-web, please send an email
+to kontakt@beberlei.de so I can send you a copy immediately.
+TXT;
+
 $rules = array(
-    'psr0' => false,
     '@PSR2' => true,
+    'array_syntax' => array(
+        'syntax' => 'long',
```
Comment (localheinz) [inline .php_cs]: Useful because of compatibility with PHP5.3.

File: .php_cs [hunk 2]
```diff
@@ -1,16 +1,78 @@
...
+    ),
+    'blank_line_after_opening_tag' => true,
+    'blank_line_before_return' => true,
+    'class_keyword_remove' => true,
```
Comment (localheinz) [inline .php_cs]: Useful because of compatibility with PHP 5.3 and PHP 5.4.

File: .php_cs [hunk 3]
```diff
@@ -1,16 +1,78 @@
...
+$header = <<<TXT
+Assert
+
+LICENSE
+
+This source file is subject to the MIT license that is bundled
+with this package in the file LICENSE.txt.
+If you did not receive a copy of the license and are unable to
+obtain it through the world-wide-web, please send an email
+to kontakt@beberlei.de so I can send you a copy immediately.
```
Comment (localheinz) [inline .php_cs]: This header comment can be adjusted to whatever it needs to be, of course.

File: .php_cs [hunk 4]
```diff
@@ -1,16 +1,78 @@
...
+TXT;
+
 $rules = array(
-    'psr0' => false,
     '@PSR2' => true,
+    'array_syntax' => array(
+        'syntax' => 'long',
+    ),
+    'blank_line_after_opening_tag' => true,
+    'blank_line_before_return' => true,
```
Comment (localheinz) [inline .php_cs]: This is the case for most `return`s in this project.

File: .php_cs [hunk 5]
```diff
@@ -1,16 +1,78 @@
...
+to kontakt@beberlei.de so I can send you a copy immediately.
+TXT;
+
 $rules = array(
-    'psr0' => false,
     '@PSR2' => true,
+    'array_syntax' => array(
+        'syntax' => 'long',
+    ),
+    'blank_line_after_opening_tag' => true,
```
Comment (localheinz) [inline .php_cs]: Reflects what we have in this project.

**Comment by rquadling** [comment]
Any idea why travis is failing for PHP7?

File: bin/generate_method_docs.php [hunk 6]
```diff
@@ -15,9 +27,9 @@ public function generateChainDocs()
 
     /**
      * @param ReflectionMethod[] $methods
-     * @param string $format
-     * @param callable|false $skipParameterTest
-     * @param string $prefix
+     * @param string             $format
```
Comment (localheinz) [inline bin/generate_method_docs.php]: I have removed the rule.

**Comment by rquadling** [comment]
Seems the blank lines before namespace and the blank line after the file level docblock is doubling up. This _could_ be a bug in PHP-CS.

**Comment by localheinz** [comment]
@rquadling 

I have boiled this PR down to only enabling the `header_comment` fixer, let me know what you think!

File: .php_cs (hunk 3 above)
Comment (rquadling) [inline .php_cs]: I've no idea which license is appropriate. I think a Creative Commons one may be simpler. From my perspective, all my code submitted to this project is CC0 - anyone can do anything they like with it. I don't know what other contributors think when they give code to GitHub projects. I know we can't change the license without @beberlei's permission.

Comment (localheinz) [inline .php_cs]: I have no idea, this is inspired by what we have, but we have headers indicating different licenses, as well as [`LICENSE`](https://github.com/beberlei/assert/blob/master/LICENSE).

**Pull request merged**